sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -815,6 +815,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_METASTORE_PARTITION_LIMIT =
+    buildConf("spark.sql.hive.metastorePartitionLimit")
+      .doc("The maximum number of metastore partitions allowed for a given table. The default " +
+           "value -1 to follow the Hive config (see HiveConf.METASTORE_LIMIT_PARTITION_REQUEST " +
+           "for more information).")
+      .version("3.1.0")
+      .intConf
+      .checkValue(_ >= -1, "The maximum must be a positive integer, -1 to follow the Hive config.")
+      .createWithDefault(100000)
```

Comment:
have you considered to keep the default behavior as it is but allow it to be configurable? changing it means now we'll need to make two HMS calls (one additional `getNumPartitionsByFilter`) which I'm not sure is desirable (have seen HMS perform very badly in production before).

Reply:
@sunchao Thank you for your response. I think this is a reasonable maximum, and Presto also has a parameter to limit the number of partitions in HiveMetadata#getPartitionsAsList, default value is 100_000

Reply:
Yea the default value 100_000 looks fine to me. My main question is whether we need to make the default value to be that and double the HMS calls. Presto doesn't call `getNumPartitionsByFilter` it seems as it streams through a partition iterator and stops once the threshold is reached.